### ISSUE TITLE
Loop detection could be done in message/term reference instead of pattern (?)

### ISSUE THREAD

**Issue by Pike** [projectfluent/fluent.js#277]
When looking at @spookylukey's evaluator, I noticed that the dirty-handling is done in pattern.

That's on a pretty hot code path.

Could we do that in message reference and term reference instead?

**Comment by stasm** [comment]
I added it in `Pattern` to make this possible:

```properties
open-foo = Open foo
    .accesskey = O
    .tooltip = Press {open-foo.accesskey} to open a Foo.
```

**Comment by Pike** [comment]
Right, it might be a bit trickier to get right. We'd need to keep `open-foo`, `open-foo[var]` and `open-foo.bar` distinct.

**Comment by spookylukey** [comment]
@Pike - and also remember the fallback logic that happens when some attributes/variants are missing. I believe I added tests for those conditions, perhaps only in my compiler branch.
